### PR TITLE
Use AbstractStore for DynamoDbSession

### PR DESCRIPTION
**Pull request by mullermp**
Uses a delegator to RackMiddleware in aws-sessionstore-dynamodb as a long term fix to missing functionality.

Also bumps the dependency to ~> 3 (which has not been released yet)

### CODE REVIEW

**Comment by mullermp** [comment]
CI is expected to be broken until aws-sessionstore-dynamodb ~> 3 is released

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,10 +1,14 @@
 Unreleased Changes
 ------------------
```
Comment (alextwoods) [inline CHANGELOG.md:3]: Nit: Mention dropping of support for rails 7.0 here.

File: lib/action_dispatch/session/dynamo_db_store.rb [hunk 2]
```diff
@@ -16,28 +16,39 @@ module Session
...
-      # generate_sid in Compatibility takes precedence over our generate_sid.
-      # Compatibility is needed so that the request is an ActionDispatch::Request.
-      CompatibilityWithoutSid = Compatibility.dup
-      CompatibilityWithoutSid.remove_method(:generate_sid)
-      include CompatibilityWithoutSid
-
-      include StaleSessionCheck
-      include SessionObject
-
+    class DynamoDbStore < ActionDispatch::Session::AbstractStore
```
Comment (alextwoods) [inline lib/action_dispatch/session/dynamo_db_store.rb:19]: Nice - I like this approach!

File: CHANGELOG.md (hunk 1 above)
Comment (mullermp) [reply CHANGELOG.md:3]: It says
* Feature - Depend on `aws-sessionstore-dynamodb ~> 3` which depends on `rack ~> 3` and is not supported by Rails `7.0`.

**Pull request merged**